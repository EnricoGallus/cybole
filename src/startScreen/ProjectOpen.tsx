import {useState} from "react";
import {useNavigate} from "react-router-dom";
import {Button, Grid, Input} from "@geist-ui/core";

function ProjectOpen() {
    const [directory, setDirectory] = useState('');
    const navigate = useNavigate();

    return (
        <Grid.Container>
            <Grid xs={24}>
                Project New
            </Grid>
            <Grid xs={24}>
                <Input value={directory} readOnly onClick={() => {
                    const dialogConfig = {
                        title: 'Select the Directory of the project',
                        buttonLabel: 'Select Directory',
                        properties: ['openDirectory']
                    };
                    window.electron.openDialog('showOpenDialog', dialogConfig)
                        .then((result: any) => setDirectory(result.filePaths[0]));
                }} />
                <Button onClick={() => {navigate('/editor', { state: { directory: directory}})}}>Open Project</Button>
            </Grid>
        </Grid.Container>
    )
}

export default ProjectOpen;