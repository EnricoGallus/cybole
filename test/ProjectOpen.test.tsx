import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProjectOpen from '../src/startScreen/ProjectOpen';
import * as tauriApi from '@tauri-apps/api/dialog';

describe('ProjectOpen', () => {
    test('renders component', () => {
        const { container } = render(<ProjectOpen />, { wrapper: MemoryRouter });

        expect(container).toMatchSnapshot();
    });

    test('openDirectoryDialog should set selected directory to input', async () => {
        jest.spyOn(tauriApi, 'open').mockImplementationOnce(() => Promise.resolve('directory'));

        render(<ProjectOpen />, { wrapper: MemoryRouter });

        const input = screen.getByLabelText('project-directory-input');

        fireEvent.click(input);

        await waitFor(() => expect(input).toHaveValue('directory'));
    });
});
