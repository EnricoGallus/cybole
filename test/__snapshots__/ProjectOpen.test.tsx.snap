// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectOpen renders component 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="col-12"
    >
      <h5>
        Open Directory and create new project
      </h5>
      <form
        class="formgroup-inline"
        novalidate=""
      >
        <div
          class="field"
        >
          <input
            class="p-inputtext p-component"
            id="projectName"
            placeholder="Project Name"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <input
            aria-label="project-directory-input"
            class="p-inputtext p-component"
            id="projectDirectory"
            placeholder="Project Directory"
            readonly=""
            value=""
          />
        </div>
        <button
          aria-label="Create Project"
          class="p-button p-component"
          type="submit"
        >
          <span
            class="p-button-label p-c"
          >
            Create Project
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
